specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json

```diff
@@ -1,2872 +1,3102 @@
-            "$ref": "./examples/CloudConnectorGetExpand.json"
-          }
-        },
-        "parameters": [
-          {
-            "$ref": "#/parameters/apiVersionParameter"
-          },
-          {
-            "$ref": "#/parameters/connectorNameParameter"
-          },
-          {
-            "name": "$expand",
-            "description": "May be used to expand the collectionInfo property. By default, collectionInfo is not included.",
-            "in": "query",
-            "required": false,
-            "type": "string"
-          }
-        ],
-        "responses": {
-          "200": {
-            "description": "OK. The request has succeeded.",
-            "schema": {
-              "$ref": "#/definitions/ConnectorDefinition"
-            }
-          },
-          "default": {
-            "description": "Error response describing why the operation failed.",
-            "schema": {
-              "$ref": "#/definitions/ErrorResponse"
-            }
-          }
-        }
-      },
-      "put": {
-        "tags": [
-          "CloudConnectors"
-        ],
-        "operationId": "CloudConnector_CreateOrUpdate",
-        "description": "Create or update a cloud connector definition",
-        "externalDocs": {
-          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
-        },
-        "x-ms-examples": {
-          "CloudConnector_Put": {
-            "$ref": "./examples/CloudConnectorPut.json"
-          }
-        },
-        "parameters": [
-          {
-            "$ref": "#/parameters/apiVersionParameter"
-          },
-          {
-            "$ref": "#/parameters/connectorNameParameter"
-          },
-          {
-            "name": "connector",
-            "in": "body",
-            "required": true,
-            "schema": {
-              "$ref": "#/definitions/ConnectorDefinition"
-            },
-            "description": "Connector details"
-          }
-        ],
-        "responses": {
-          "200": {
-            "description": "OK. The request has succeeded.",
-            "schema": {
-              "$ref": "#/definitions/ConnectorDefinition"
-            }
-          },
-          "201": {
-            "description": "Created.",
-            "schema": {
-              "$ref": "#/definitions/ConnectorDefinition"
-            }
-          },
-          "default": {
-            "description": "Error response describing why the operation failed.",
-            "schema": {
-              "$ref": "#/definitions/ErrorResponse"
-            }
-          }
-        }
-      },
-      "delete": {
-        "tags": [
-          "CloudConnectors"
-        ],
-        "operationId": "CloudConnector_Delete",
-        "description": "Delete a cloud connector definition",
-        "externalDocs": {
-          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
-        },
-        "x-ms-examples": {
-          "CloudConnector_Delete": {
-          
```

Comment:
Defining this as an open object prevents client tooling from validating/assisting proper input. Have a look at [this swagger](https://github.com/Azure/azure-rest-api-specs/blob/672c412811930cefa3c02143f342ce493f7b154e/specification/security/resource-manager/Microsoft.Security/preview/2015-06-01-preview/externalSecuritySolutions.json) to see use of the "discriminator" concept. In essence, you define a "discriminator" property based on the enum `ruleType` and define three definitions for this object. On the new object definitions, you would set a `x-ms-discriminator-valuex-ms-discriminator-value` property to match one of the three enum values.

Reply:
done